src/components/VeggyBody/VeggyProductItem/ProductItem.jsx

```diff
@@ -0,0 +1,120 @@
+import { useState } from "react";
+import './ProductItem.css';
+
+export default function ProductItem (props) {
+    const {
+        product,
+        selectedProducts,
+        setSelectedProducts,
+        zoomProductImage,
+        setShakingCart,
+    } = props;
+
+    const [count, setCount] = useState(1);
+    const [buttonText, setButtonText] = useState('ADD TO CART');
+
+    function handleClickChangeButtonText() {
+        setButtonText('ADDED');
+
+        setTimeout(() => {
+            setButtonText('ADD TO CART');
+        }, 1000);
+    }
+
+    const addCountHandler = () => {
+        setCount(n => n + 1);
+    };
+
+    const removeCountHandler = () => {
+        if(count === 1){
+            alert('Your counter cannot be less than 0');
+            return;
+        }
+        setCount(count - 1);
+    };
+
+    function checkInputCounter (e) {
```

Comment:
```suggestion
    function handleSetProductQuantity(e) {
```
чому check? Ми в цій функції не перевіряємо кількість продукту, а банально задаєм його кількість.